src/Compilers/CSharp/Portable/Binder/BinderFactory.BinderFactoryVisitor.cs

```diff
@@ -795,22 +793,31 @@ internal InContainerBinder VisitCompilationUnit(CompilationUnitSyntax compilatio
                         //
                         // Binder chain in script/interactive code:
                         //
-                        // + global usings
-                        //   + interactive usings (in an interactive session)
+                        // + global imports
+                        //   + current and previous submission imports (except using aliases)
                         //     + global namespace
                         //       + host object members
-                        //         + previous submissions (in an interactive submission)
-                        //           + script class members & top-level using aliases
+                        //         + previous submissions and corresponding using aliases
+                        //           + script class members and using aliases
                         //
 
-                        if (compilation.GlobalImports.Usings.Length > 0)
+                        // This is declared here so it can be captured.  It's initialized below.
+                        InContainerBinder scriptClassBinder = null;
+
+                        if (inUsing)
                         {
-                            result = new UsingsBinder(result, compilation.GlobalImports.Usings);
+                            result = result.WithAdditionalFlags(BinderFlags.InScriptUsing);
                         }
-
-                        if (compilation.IsSubmission)
+                        else
                         {
-                            result = new InteractiveUsingsBinder(result);
+                            result = new InContainerBinder(container: null, next: result, imports: compilation.GlobalImports);
+
+                            // NB: This binder has a full Imports object, but only the non-alias imports are
+                            // ever consumed.  Aliases are actually checked in scriptClassBinder (below).
+                            result = compilation.PreviousSubmission == null
+                                ? new InContainerBinder(result, basesBeingResolved => scriptClassBinder.GetImports(basesBeingResolved))
+                                : new InContainerBinder(result, basesBeingResolved =>
+                                    compilation.PreviousSubmissionImports.Concat(scriptClassBinder.GetImports(basesBeingResolved)));
```

Comment:
Concat concatenates current imports to previous imports. Does that mean we'll choose the extern alias from the previous submission if there are duplicates with the same name but different target?